hypershift-operator/controllers/nodepool/conditions.go

```diff
@@ -852,3 +854,103 @@ func (r NodePoolReconciler) validPlatformConfigCondition(ctx context.Context, no
+		// Use the desired version as the control plane version
+		controlPlaneVersionStr = hcluster.Status.Version.Desired.Version
+	} else {
+		// If the cluster is installed or upgrading
+		// Start with the most recent version from history as the default
+		controlPlaneVersionStr = hcluster.Status.Version.History[len(hcluster.Status.Version.History)-1].Version
+		// Update with any more recent Completed version if found
+		for _, history := range hcluster.Status.Version.History {
+			if history.State == "Completed" {
+				controlPlaneVersionStr = history.Version
+				break
+			}
+		}
+	}
+	// Parse control plane version
+	controlPlaneVersion, err := semver.Parse(controlPlaneVersionStr)
+	if err != nil {
+		return &ctrl.Result{}, fmt.Errorf("parsing control plane version (%s): %w", controlPlaneVersionStr, err)
+	}
+
+	releaseImage, err := r.getReleaseImage(ctx, hcluster, nodePool.Status.Version, nodePool.Spec.Release.Image)
+	if err != nil {
+		SetStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
+			Type:               hyperv1.NodePoolSupportedVersionSkewConditionType,
+			Status:             corev1.ConditionUnknown,
+			Reason:             hyperv1.NodePoolValidationFailedReason,
+			Message:            fmt.Sprintf("Failed to get release image: %v", err.Error()),
+			ObservedGeneration: nodePool.Generation,
+		})
+		return &ctrl.Result{}, fmt.Errorf("failed to look up release image metadata: %w", err)
+	}
+
+	// Parse NodePool version
+	nodePoolVersion, err := semver.Parse(releaseImage.Version())
+	if err != nil {
+		return &ctrl.Result{}, fmt.Errorf("parsing NodePool version (%s): %v", releaseImage.Version(), err)
+	}
+
+	// NodePool version cannot be higher than control plane version
+	if nodePoolVersion.GT(controlPlaneVersion) {
+		SetStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
+			Type:   hyperv1.NodePoolSupportedVersionSkewConditionType,
+			Status: corev1.ConditionFalse,
+			Reason: hyperv1.NodePoolUnsupportedSkewReason,
+			Message: fmt.Sprintf("NodePool version %s cannot be higher than the HostedCluster version %s",
+				nodePoolVersion, controlPlaneVersion),
+			ObservedGeneration: nodePool.Generation,
+		})
+		return nil, nil
+	}
+
+	// Calculate minor version difference
+	versionDiff := int64(controlPlaneVersion.Minor - nodePoolVersion.Minor)
+
+	// For 4.even versions, allow y-2 difference
+	// For 4.odd versions, allow y-1 difference
+	maxAllowedDiff := int64(2)
+	if controlPlaneVersion.Minor%2 == 1 {
+		maxAllowedDiff = 1
+	}
+
+	if versionDiff > maxAllowedDiff {
+		SetStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
+			Type:   hyperv1.NodePoolSupportedVersionSkewConditionType,
+			Status: corev1.ConditionFalse,
+			Reason: hyperv1.NodePoolUnsupportedSkewReason,
+			Message: fmt.Sprintf("NodePool minor version %d.%d is not compatible with the HostedCluster minor version %d.%d (max allowed difference: %d)",
```

Comment:
nit: this generates messages like:

> NodePool minor version 4.15 is not compatible with the HostedCluster minor version 4.17 (max allowed difference: 1)

But that doesn't explain exactly how the difference integer fits in.  It might be easier to read if we construct the minimum allowed 4.y, like:

```golang
fmt.Sprintf("NodePool minor version %d.%d is less than %.%, which is the minimum NodePool version compatible with the %d.%d HostedCluster",
	nodePoolVersion.Major, nodePoolVersion.Minor,
	controlPlaneVersion.Major, controlPlaneVersion.Minor - maxAllowedDiff,
	controlPlaneVersion.Major, controlPlaneVersion.Minor)
```

to generate messages like:

> NodePool minor version 4.15 is less than 4.16, which is is the minimum NodePool version compatible with the 4.17 HostedCluster